Bio/SeqIO/NibIO.py

```diff
@@ -160,6 +159,14 @@ def write_file(self, records):
             raise ValueError("Sequence should contain A,C,G,T,N,a,c,g,t,n only")
         indices = nucleotides.translate(table)
         handle.write(bytes.fromhex(indices))
+
+    def write_file(self, records):
+        """Use this to write an entire file containing the given records."""
+        count = super().write_file(records)
+        if count == 0:
+            raise ValueError("Must have one sequence")
+        if count > 1:
+            raise ValueError("More than one sequence found")
```

Comment:
This seems raising an error after the fact - you might have just written a 1000 records giving a malformed NIB file. Surely better to fail immediately? A try/except with ``next()`` to confirm just one record seems better here.

Reply:
Yes. But the best way to solve this is in the base class, which has its own problems. We'll get to that, don't worry.

Reply:
The main base class assumes multiple records, a subclass for single record formats would be a good general solution?

Reply:
Possibly. I am looking for a solution where only a single class holds the `handle` as an attribute.

Reply:
That sounds compatible.